### PR TITLE
Starter code

### PR DESCRIPTION
**Pull request by CheezItMan**
Setup starter code.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,26 +1,68 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { useState } from 'react';
 import './App.css';
 
-function App() {
+import Board from './components/Board';
+
+const PLAYER_1 = 'x';
+const PLAYER_2 = 'o';
```
Comment (kaidamasaki) [inline src/App.js]: Capital letters for extra intensity?

```suggestion
const PLAYER_1 = 'X';
const PLAYER_2 = 'O';
```

File: src/App.js [hunk 2]
```diff
@@ -1,26 +1,68 @@
...
+      });
+      col += 1;
+      currentId += 1;
+    }
+    row += 1;
+    col = 0;
+  }
+
+  return squares;
+}
```
Comment (kaidamasaki) [inline src/App.js]: It's more idiomatic to use `for` loops here.

```suggestion
const generateSquares = () => {
  const squares = [];
  let currentId = 0;
  console.log(`row = ${ row }`);

  for (let row = 0; row < 3; row++) {
    squares.push([]);
    while (let col = 0; col < 3; col++) {
      squares[row].push({
        id: currentId,
        value: '',
      });
      currentId += 1;
    }
  }

  return squares;
}
```

File: src/components/Board.css [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+	height: 600px;
+	margin: 0 auto;
+  background-color: #34495e;
+  color: #fff;
+  border: 6px solid #2c3e50;
+  border-radius: 10px;
+	
+	display: grid;
+	grid-template: repeat(3, 1fr) / repeat(3, 1fr);
+}
```
Comment (kaidamasaki) [inline src/components/Board.css]: Funky indentation (tabs + spaces): 

```suggestion
div.grid {
    width: 600px;
    height: 600px;
    margin: 0 auto;
    background-color: #34495e;
    color: #fff;
    border: 6px solid #2c3e50;
    border-radius: 10px;
    display: grid;
    grid-template: repeat(3, 1fr) / repeat(3, 1fr);
}
```

File: src/components/Board.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+
+const generateSquares = (squares, onClickCallback) => {
+  // Complete this for Wave 1
+
+}
+
+const Board = ({ squares, onClickCallback }) => {
+
+  const squareList = generateSquares(squares, onClickCallback);
```
Comment (kaidamasaki) [inline src/components/Board.js]: Extraneous newline:

```suggestion
const Board = ({ squares, onClickCallback }) => {
  const squareList = generateSquares(squares, onClickCallback);
```

File: src/components/Board.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  </div>
+}
+
+Board.propTypes = {
+  squares: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.shape({
+    id: PropTypes.number.isRequired,
+    value: PropTypes.string.isRequired
+  }))),
+  onClickCallback: PropTypes.func.isRequired,
+};
```
Comment (kaidamasaki) [inline src/components/Board.js]: I'd split this out for better readability:

```suggestion
Board.propTypes = {
  squares: PropTypes.arrayOf(
    PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        value: PropTypes.string.isRequired
      })
    )
  ),
  onClickCallback: PropTypes.func.isRequired,
};
```

File: src/components/Square.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+const Square = (props) => {
+  // For Wave 1 enable this 
+  //  Component to alert a parent 
+  //  component when it's clicked on.
+
+  return <button
+    className="square"
+
+  >
```
Comment (kaidamasaki) [inline src/components/Square.js]: I suspect you're leaving room for additional class names but it makes this hard to parse:

```suggestion
    className="square"
  >
```

**Review by kaidamasaki — COMMENTED** [review]
Looks pretty good.  Just a few small things I'd change.

File: src/App.js [hunk 7]
```diff
@@ -1,26 +1,68 @@
...
+
+const App = () => {
+
+  const [squares, setSquares] = useState(generateSquares());
+
+  // Wave 1
+  // You will need to create a method to change the square 
+  //   When it is clicked on.
+  //   Then pass it into the squares as a callback
```
Comment (tildeee) [inline src/App.js:35]: After the README is in place, will the phrase "Then pass it into the squares as a callback" make sense? The thing that confuses me is the phrase "into the squares," is that talking about line 60?

**Review by tildeee — APPROVED** [review]
Only other thought I might have is that by this point, they don't know tests, so it might make sense to delete the test files. Not necessary though.

File: src/components/Board.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import './Board.css';
+import Square from './Square';
+import PropTypes from 'prop-types';
+
+
+const generateSquares = (squares, onClickCallback) => {
```
Comment (jmaddox19) [inline src/components/Board.js]: this is the same name of the fucntion in `App`. Would you wanna rename this to something else? I'm struggling to think of a more fitting name... ugh

**Review by jmaddox19 — APPROVED** [review]
Thanks Chris!

File: src/App.js (hunk 7 above)
Comment (jmaddox19) [reply src/App.js:35]: I'm with Dee that this feels a little out of place here.

File: src/components/Board.js (hunk 8 above)
Comment (jmaddox19) [reply src/components/Board.js]: Thanks! Love it

File: src/App.js [hunk 9]
```diff
@@ -1,26 +1,63 @@
...
+    for (let col = 0; col < 3; col += 1) {
+      squares[row].push({
+        id: currentId,
+        value: '',
+      });
+      currentId += 1;
+    }
+  }
+
+  return squares;
```
Comment (jmaddox19) [inline src/App.js:25]: Adding this comment here from the other PR since it feels more relevant here:

Do we want to update the starter code to just create an array of 9 squares since we are relying on styling to actually dictate the number of squares per row anyway?
It just seems a little funky to me to give them an array of arrays that they immediately flatten and do nothing meaningful with the nesting of the original array.

File: src/App.js (hunk 7 above)
Comment (jmaddox19) [reply src/App.js:35]: Can we move it to line 60?
Also, I see now that you update the wave number here. Thank you! lines 39 and 44 need updated wave numbers as well. (If only we could set them to auto-increment when a list insertion happens!)

File: src/App.js (hunk 9 above)
Comment (CheezItMan) [reply src/App.js:25]: I kept it a 2D array because it's much easier to conceptually figure out a winner this way.  Maybe I'm wrong and feel free to overrule me.
